### PR TITLE
Expand Detector for CLI language detection

### PR DESCRIPTION
**Pull request by robertjli**
### What does this PR do?

* Add a `Process` interface that exposes fields necessary to detect language with either CLI or binary analysis
* Update the `Detector` interface to accept the `Process` interface so that it can be used for either analysis type
* Update `GoDetector` to not use pointer receivers

### Motivation

This allows us to generalize the different language detection calls.

### Describe how to test/QA your changes

Verify that language detection features work as intended, with no regressions.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by just-chillin — APPROVED** [review]

File: pkg/languagedetection/detector.go
Comment (robertjli) [inline pkg/languagedetection/detector.go:1]: Moved to `languagemodels/detector.go` due to dependency cycle

File: cmd/system-probe/modules/language_detection.go [hunk 1]
```diff
@@ -88,7 +80,12 @@ func detectLanguage(writer http.ResponseWriter, request *http.Request) {
 		return
 	}
 
-	resp := toDetectLanguageResponse(languagedetection.DetectWithPrivileges(getRequestPids(&req)))
+	procs := make([]languagemodels.Process, 0, len(req.Processes))
+	for _, proc := range req.Processes {
+		procs = append(procs, proc)
+	}
```
Comment (brycekahle) [inline cmd/system-probe/modules/language_detection.go:86]: suggestion: you will need to import "golang.org/x/exp/slices" for this.
```suggestion
	procs := slices.Clone(req.Processes)
```

**Review by brycekahle — APPROVED** [review]

File: cmd/system-probe/modules/language_detection.go (hunk 1 above)
Comment (robertjli) [reply cmd/system-probe/modules/language_detection.go:86]: The reason for the slice copy is to change the type from the [proto impl type](https://github.com/DataDog/datadog-agent/blob/aea395a14e706609ea175ca85d0aa4bf710ed058/pkg/proto/datadog/languagedetection/api.proto#L7-L11) to the [interface `languagemodels.Process`](https://github.com/DataDog/datadog-agent/blob/aea395a14e706609ea175ca85d0aa4bf710ed058/pkg/languagedetection/languagemodels/detector.go#L14), I'm not sure if we can achieve that with `slices.Clone()`?

Comment (brycekahle) [reply cmd/system-probe/modules/language_detection.go:86]: Yeah, I don't think you can.

**Pull request merged**